src/shared/driverUtils.js

```diff
@@ -0,0 +1,258 @@
+ * @return {Object} An object that has the status of the last request regardless
+ *                  of success, and the node ChildProcess started by `exec()`
+ *                  or `null` if it failed to start.
+ */
+fluid.personalData.startServer = async function (serverPort) {
+    const url = "http://localhost:" + serverPort;
+    let resp = {};
+
+    console.debug(`- Checking if server is running using ${url}`);
+    try {
+        resp = await axios(url);
+        return { status: resp.status, process: null, wasRunning: true };
+    }
+    catch (error) { ; }
+
+    // Server needs to be started
+    console.debug(`- Starting server at ${url}`);
+    const pdServerProcess = exec(fluid.personalData.startServerCmd + " " + serverPort);
+    for (let i = 0; i < NUM_CHECK_REQUESTS; i++) {
+        try {
+            console.debug(`... attempt ${i}`);
+            resp = await axios(url);
+        }
+        catch (error) {
+            console.debug(`... attempt ${i} error: ${error.message}`);
+            resp.status = 503;
+        }
+        if (resp.status === 200) {
+            break;
+        } else {
+            console.debug(`... attempt ${i} going to sleep`);
+            await fluid.personalData.sleep(DELAY);
+        }
+    }
+    console.debug(`- Attempt to start server, result:  ${resp.status}`);
+    return { status: resp.status, process: pdServerProcess, wasRunning: false };
+};
+
+/**
+ * Stop the personal data server, if it was started by this testing sequence.
+ * If it was started in order to run the tests, the `childProcess` argument will
+ * have a non-null ChildProcess instance and a `wasRunning` flag set to false.
+ * If so, check the ChildProcess's exit status and, if it has not exited, invoke
+ * its `kill()` method.
+ *
+ * @param {Object} childProcessInfo - Information about the server's process
+ * @param {Object} childProcessInfo.wasRunning - True if the server was already
+ *                                               running.
+ * @param {Object} childProcessInfo.process - ChildProcess instance `wasRunning`
+ *                                            created if the server was not
+ *                                            running.
+ * @param {String} url - Url to use to check if the ChildProcess has been killed.
+ */
+fluid.personalData.stopServer = async function (childProcessInfo, url) {
+    console.debug(`- Stopping server at ${url}`);
+    if (childProcessInfo.wasRunning === false && childProcessInfo.process &&
+        childProcessInfo.process.exitCode === null) {
+        childProcessInfo.process.kill();
+    }
+    // Wait for the childProcess to fully exit
+    for (let i = 0; i < NUM_CHECK_REQUESTS; i++) {
+        try {
+            console.debug(`... attempt ${i}`);
+            await axios(url);
+        }
+        catch (error) {
+            break;
+        }
+        await fluid.personalData.sleep(DELAY);
+    }
+};
```

Comment:
A normal exit to this function when we reach NUM_CHECK_REQUESTS should represent an error, since in our opinion, we will have failed to stop the server.